compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs

```diff
@@ -795,14 +795,18 @@ impl<'tcx> Constructor<'tcx> {
             (Variant(self_id), Variant(other_id)) => self_id == other_id,
 
             (IntRange(self_range), IntRange(other_range)) => self_range.is_covered_by(other_range),
-            (
-                FloatRange(self_from, self_to, self_end),
-                FloatRange(other_from, other_to, other_end),
-            ) => {
-                match (
-                    compare_const_vals(pcx.cx.tcx, *self_to, *other_to, pcx.cx.param_env),
-                    compare_const_vals(pcx.cx.tcx, *self_from, *other_from, pcx.cx.param_env),
-                ) {
+            (F32Range(self_from, self_to, self_end), F32Range(other_from, other_to, other_end)) => {
+                match (self_to.partial_cmp(other_to), self_from.partial_cmp(other_from)) {
+                    (Some(to), Some(from)) => {
+                        (from == Ordering::Greater || from == Ordering::Equal)
+                            && (to == Ordering::Less
+                                || (other_end == self_end && to == Ordering::Equal))
+                    }
+                    _ => false,
+                }
```

Comment:
Can this condition be simplified a bit? It took me a bit of time to read it.
```suggestion
                self_from.ge(other_from) && match self_to.partial_cmp(other_to) {
                    Some(Ordering::Less) => true,
                    Some(Ordering::Equal) => other_end == self_end,
                    _ => false,
                }
```

Reply:
This is much better ty :D